### ISSUE TITLE
Ubuntu 16.04: using backwards-compatible default setting chroot=y

### ISSUE THREAD

**Issue by igalic** [camptocamp/puppet-postfix#137]
Ubuntu 16.04 comes with postfix 3.1.0 which.
Postfix 3.0 has changed the [default for chroot](http://www.postfix.org/master.5.html):

> Chroot (default: Postfix >= 3.0: n, Postfix <3.0: y)

For an easy fix, we should simply be explicit as to which daemons should be chrooted, while following the advise on the same page:

>               Chroot should not be used with the local(8), pipe(8),  spawn(8),
              and virtual(8) daemons.  Although the proxymap(8) server can run
              chrooted, doing so defeats most of the purpose  of  having  that
              service in the first place.

**Comment by raphink** [comment]
So you're suggesting we should use false for Ubuntu 16.04?

**Comment by igalic** [comment]
i honestly don't know
it's really complicated, or, else depends on whether you wanna keep compatibility with the module or the software

**Comment by raphink** [comment]
Yes. I'd rather the module behave in a standard manner as much as possible.

**Comment by amateo** [comment]
Maybe you could include PR #170 that adds a parameter to configure chroot. If you leave it undefined, the it uses the software default (```-``` in _master.cf_ ) but you could also force it to true (```y```) or false (```n```).
But the PR is failing because of problems, I think that not directly related with the PR, building ubuntu machines for CI.

**Comment by raphink** [comment]
Would https://github.com/camptocamp/puppet-postfix/pull/224 fix this bug?

**Comment by lelutin** [comment]
hello! from what I can see, #224 changes the default value of `$chroot` to false for all distributions.

debian uses chrooted daemons by default (which postfix upstream is not too happy about, but that's another story)

so the solution here depends on you goals towards how things are managed:

if you'd like for the puppet module to have default settings that are as close as possible to distro defaults, then you could special-case the value of the `$chroot` parameter in the params class.

however, if you want to simplify and/or standardize how the module manages default configuration, then #224 does this well.

In all cases, the `undef` value to `$chroot` seems useless after the change, so it'd probably be nice to add a deprecation warning and remove it at some later point to keep only 'y' and 'n' as possible values.

**Comment by raphink** [comment]
@lelutin 	if the module was a new one, I'd say we should be as close as possible to the OS standard. However because this module already has a history which is to set chroot to false by default, I think we should keep it backwards-compatible.

**Comment by lelutin** [comment]
@raphink this makes a lot of sense!

**Comment by raphink** [comment]
Merged  #224 which makes all chroot options default to false as it's always been in this module.
